ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/Queries.scala

```diff
@@ -720,7 +720,7 @@ private final class PostgresQueries(tablePrefix: String, tpIdCacheMaxEntries: Lo
       s"""
         INSERT INTO $contractTableNameRaw
         VALUES (?, ?, ?::jsonb, ?, ?::jsonb, ?, ?, ?)
-        ON CONFLICT (contract_id) DO NOTHING
+        ON CONFLICT DO NOTHING
```

Comment:
this might be too relaxed. If we insert a different contract with the same key, this will also ignore it but we should probably throw an error instead.

Reply:
Not sure. The current implementation would let through the case in which we insert a different contract with the same contract identifier. In both cases, this is a ledger error, and I'm not sure we should fail. The failure should be caught earlier on. At this point, the worst thing that happens is that you corrupt a cache that you can throw away. Makes sense?